From 777fca3d7d77da52e94cc3410d755920a9eb221a Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:41:38 -0400
Subject: [alsa] hda/hdmi: apply Valleyview fix-ups to Cherryview display codec

Message-id: <140972823573-428-patchset-jkysela@redhat.com>
Patchwork-id: 91124
O-Subject: [RHEL 7.1 PATCH 427/442] ALSA: hda/hdmi - apply Valleyview fix-ups to Cherryview display codec
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Valleyview and Cherryview have the same behavior on display audio. So this patch
defines is_valleyview_plus() to include codecs for both Valleyview and its successor
Cherryview, and apply Valleyview fix-ups to Cherryview.

Signed-off-by: Libin Yang <libin.yang@intel.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: ca2e7224d7e7d424e69616634f90f3f428710085)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_hdmi.c b/sound/pci/hda/patch_hdmi.c
index 5e229f7..99d7d7f 100644
--- a/sound/pci/hda/patch_hdmi.c
+++ b/sound/pci/hda/patch_hdmi.c
@@ -50,6 +50,8 @@ MODULE_PARM_DESC(static_hdmi_pcm, "Don't restrict PCM parameters per ELD info");
 #define is_haswell_plus(codec) (is_haswell(codec) || is_broadwell(codec))
 
 #define is_valleyview(codec) ((codec)->vendor_id == 0x80862882)
+#define is_cherryview(codec) ((codec)->vendor_id == 0x80862883)
+#define is_valleyview_plus(codec) (is_valleyview(codec) || is_cherryview(codec))
 
 struct hdmi_spec_per_cvt {
  hda_nid_t cvt_nid;
@@ -1459,7 +1461,7 @@ static int hdmi_pcm_open(struct hda_pcm_stream *hinfo,
        mux_idx);
 
  /* configure unused pins to choose other converters */
- if (is_haswell_plus(codec) || is_valleyview(codec))
+ if (is_haswell_plus(codec) || is_valleyview_plus(codec))
   intel_not_share_assigned_cvt(codec, per_pin->pin_nid, mux_idx);
 
  snd_hda_spdif_ctls_assign(codec, pin_idx, per_cvt->cvt_nid);
@@ -1598,7 +1600,8 @@ static bool hdmi_present_sense(struct hdmi_spec_per_pin *per_pin, int repoll)
    *   and this can make HW reset converter selection on a pin.
    */
   if (eld->eld_valid && !old_eld_valid && per_pin->setup) {
-   if (is_haswell_plus(codec) || is_valleyview(codec)) {
+   if (is_haswell_plus(codec) ||
+    is_valleyview_plus(codec)) {
     intel_verify_pin_cvt_connect(codec, per_pin);
     intel_not_share_assigned_cvt(codec, pin_nid,
        per_pin->mux_idx);
@@ -1779,7 +1782,7 @@ static int generic_hdmi_playback_pcm_prepare(struct hda_pcm_stream *hinfo,
  bool non_pcm;
  int pinctl;
 
- if (is_haswell_plus(codec) || is_valleyview(codec)) {
+ if (is_haswell_plus(codec) || is_valleyview_plus(codec)) {
   /* Verify pin:cvt selections to avoid silent audio after S3.
    * After S3, the audio driver restores pin:cvt selections
    * but this can happen before gfx is ready and such selection
@@ -2330,7 +2333,7 @@ static int patch_generic_hdmi(struct hda_codec *codec)
   intel_haswell_fixup_enable_dp12(codec);
  }
 
- if (is_haswell_plus(codec) || is_valleyview(codec))
+ if (is_haswell_plus(codec) || is_valleyview_plus(codec))
   codec->depop_delay = 0;
 
  if (hdmi_parse_codec(codec) < 0) {
-- 
1.7.1